libethereum/EthereumPeer.cpp

```diff
@@ -37,8 +37,10 @@ using namespace dev;
 using namespace dev::eth;
 using namespace p2p;
 
-const unsigned c_maxIncomingNewHashes = 1024;
-const unsigned c_maxHashesToSend = 65536;
+static const unsigned c_maxIncomingNewHashes = 1024;
+static const unsigned c_maxHeadersToSend = 65536;
```

Comment:
Should be much lower, I suggest 1024. Header size is considerably larger than hash. Also, there is 16mb packet size limit.